### ISSUE TITLE
Wan Training on Videos

### ISSUE THREAD

**Issue by DavidsenT** [tdrussell/diffusion-pipe#130]
Hey, 

So it would be great to have a bit more directions regarding Loras on Wan. Currently trying to achieve a high quality training, I've done many tests with different video framerates and frame lenght. 

My original thinking is that as Wan is 16fps, my dataset should too. I've made my videos 16n+1 frames long (17 33 49 65 81 97) and after caching, I get 

`[2025-03-04 09:28:15,294] [WARNING] [dataset.py:174:_make_divisible_by] size bucket (1024, 1024, 17) is being completely dropped because it doesn't have enough images
[2025-03-04 09:28:15,294] [WARNING] [dataset.py:174:_make_divisible_by] size bucket (1024, 1024, 49) is being completely dropped because it doesn't have enough images
[2025-03-04 09:28:15,294] [WARNING] [dataset.py:174:_make_divisible_by] size bucket (1024, 1024, 97) is being completely dropped because it doesn't have enough images
[2025-03-04 09:28:15,294] [WARNING] [dataset.py:174:_make_divisible_by] size bucket (1024, 1024, 81) is being completely dropped because it doesn't have enough images
[2025-03-04 09:28:15,294] [WARNING] [dataset.py:174:_make_divisible_by] size bucket (1024, 1024, 33) is being completely dropped because it doesn't have enough images
[2025-03-04 09:28:15,295] [WARNING] [dataset.py:174:_make_divisible_by] size bucket (1024, 1024, 65) is being completely dropped because it doesn't have enough images` 

And here is the corresponding line in the dataset.toml : frame_buckets = [1, 17, 33, 49, 65, 81, 97]

I've tried to load 48 frame long videos at 24fps (which worked perfectly on Hunyuan) and it only process these as 24 frames long. 

Is there a magic formula to know regarding what either the frame lenght should be OR what number should be put in the frame bucket for Wan lora trainings ? 

Also quick question regarding the overall video processing during dataset caching. I've tried (as a test) to throw a bunch of random clips, different lenghts etc without processing these. And to my understanding, it create a clip of the maximum possible lenght from the frame bucket values (ie, if my clip is 200+ frame long, and my max value in my frame bucket is 97, it creates a 97 clip) is that right ? 

Anyway, sorry for the rambling but I'm open to any tips and tricks to understand how to process my videos to train Wan 14b. 

Thanks !

**Comment by luojunjie2020** [comment]
Same problem on my end. Any solutions?

**Comment by thomaswinged** [comment]
I was about to ask the same question—have you found the answer yet?

**Comment by tdrussell** [comment]
I think there are 2 main things to understand here:
1. It will assign source videos to the longest frame bucket it can, such that the video is still longer than the frame bucket length. Then, clips of exactly the frame bucket length are extracted according to the video_clip_mode setting.
2. Each final size bucket must be made divisible by the global batch size, and it does this by rounding down (dropping images/videos). E.g. 4 GPUs in a data parallel configuration, batch size per GPU 2, and 4 gradient accumulation steps, equals a 32 global batch size. Any size bucket having less than 32 items gets rounded down to 0 and that log line is printed. If every video bucket is logged as being dropped, either all videos are shorter than the shortest frame bucket (sounds like this is not the case), your global batch size is high enough that the above mentioned rounding is dropping everything, or some bug is happening and something is going wrong.

Also you can make your source video dataset 16 FPS, and that might be slightly higher quality that way, but it's not strictly necessary. The code will load the video into the FPS of the model, by adding / dropping frames as needed.

**Comment by thomaswinged** [comment]
I just launched training with a mix of videos and images, using the setting `frame_buckets = [1, 17, 49]`. With images and videos (25 FPS) that are 17 or 49 frames long, everything is working fine, and the training has started successfully. However, when I tried to include 81-frame videos, I encountered an error due to insufficient VRAM (I'm using an RTX A6000 with 48GB). My dataset is sized at 512x512.

**Comment by DavidsenT** [comment]
I think more frames in a video = more vram 

So basically working it 16fps is the way to go to have a proper dataset meant for Wan, using other fps will inherently drop frames and change the frame count. And, let's say i'm using a batch of 16 (4 micro batch 4 gradient) I need at least 16 clips of each frame lenght to have that bucket working. 

I just wonder, is there any upside in using different buckets or only using 48 frames for example is not going to do anything less than using 3 or 4 different buckets ?

**Comment by aungkhant0911** [comment]
@tdrussell 

> 1. It will assign source videos to the longest frame bucket it can, such that the video is still longer than the frame bucket length. Then, clips of exactly the frame bucket length are extracted according to the video_clip_mode setting.

I am wondering if it is possible for you to provide another related option (eg; skip_frame =3) on how frames of a clip, which is to be extracted from a video, are sampled. Eg; lets say I have a video with 64 frames long, and I want to exactly extract one clip from that. So if skip_frame = 0, then the very first X consecutive frames are extracted. If skip_frame = 2, every 3rd frame is sampled for a total of X frames. skip_frame = -1 means (video / X) frame is skipped, etc...

**Comment by Jalen-Brunson** [comment]
I’ve had the bucket warning but it wasn’t related to length of frames, it was the resolution of the videos.  I had a 50+ video data set, all varying lengths.  Most of the videos were 1280x720 but I had 3 videos that were 480 and 4 was the minimum required.  I modified the dataset.py file to show me what files were causing the issue,  but ultimately it came down to my data not being normalized.  

He lengths of many of my videos exceeded 81 frames, but I was using dual H200s.

**Comment by olehswift** [comment]
Should I preprocess the dataset by removing or masking the background (e.g. using a white or transparent BG) to get cleaner and more realistic LoRA results?

**Comment by Jalen-Brunson** [comment]
I think you actually want the background for a character Lora.  I’ve gotten pretty realistic results for character and action loras without masking.
